File: smash-ranks-android/app/src/test/java/com/garpr/android/misc/TimberTest.java
package com.garpr.android.misc;

import com.garpr.android.BaseTest;

import org.junit.Before;
import org.junit.Test;

import java.util.List;

import javax.inject.Inject;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class TimberTest extends BaseTest {

    private static final String TAG = "TimberTest";

    @Inject
    Timber mTimber;


    @Before
    @Override
    public void setUp() throws Exception {
        super.setUp();
        getTestAppComponent().inject(this);
    }

    @Test
    public void testClearEntries() throws Exception {
        mTimber.clearEntries();
        List<Timber.Entry> entries = mTimber.getEntries();
        assertTrue(entries.isEmpty());

        mTimber.d(TAG, "blah");
        mTimber.clearEntries();
        entries = mTimber.getEntries();
        assertTrue(entries.isEmpty());
    }

    @Test
    public void testGetEntries() throws Exception {
        List<Timber.Entry> entries = mTimber.getEntries();
        assertTrue(entries.isEmpty());
    }

    @Test
    public void testGetEntriesWithOne() throws Exception {
        mTimber.d(TAG, "one");
        final List<Timber.Entry> entries = mTimber.getEntries();
        assertEquals(entries.size(), 1);
    }

    @Test
    public void testGetEntriesWithTwo() throws Exception {
        mTimber.d(TAG, "one");
        mTimber.w(TAG, "two");
        final List<Timber.Entry> entries = mTimber.getEntries();
        assertEquals(entries.size(), 2);
    }

    @Test
    public void testGetEntriesWithThree() throws Exception {
        mTimber.d(TAG, "one");
        mTimber.w(TAG, "two");
        mTimber.e(TAG, "three");
        final List<Timber.Entry> entries = mTimber.getEntries();
        assertEquals(entries.size(), 3);
    }

}

File: smash-ranks-android/app/src/test/java/com/garpr/android/misc/TimberTest.kt
package com.garpr.android.misc

import com.garpr.android.test.BaseTest
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import org.koin.test.inject
import kotlin.random.Random

class TimberTest : BaseTest() {

    protected val timber: Timber by inject()

    @Test
    fun testClearEntries() {
        timber.clearEntries()

        var entries = timber.entries
        assertTrue(entries.isEmpty())

        timber.d(TAG, "blah")
        timber.clearEntries()
        entries = timber.entries
        assertTrue(entries.isEmpty())
    }

    @Test
    fun testGetEntries() {
        val entries = timber.entries
        assertTrue(entries.isEmpty())
    }

    @Test
    fun testGetEntriesWithOne() {
        timber.d(TAG, "one")
        val entries = timber.entries
        assertEquals(1, entries.size)
    }

    @Test
    fun testGetEntriesWithTwo() {
        timber.d(TAG, "one")
        timber.w(TAG, "two")
        val entries = timber.entries
        assertEquals(2, entries.size)
    }

    @Test
    fun testGetEntriesWithThree() {
        timber.d(TAG, "one")
        timber.w(TAG, "two")
        timber.e(TAG, "three")
        val entries = timber.entries
        assertEquals(3, entries.size)
    }

    @Test
    fun testMaxSize() {
        val times = Random.nextInt(1000, 2000)

        repeat(times) {
            timber.d(TAG, it.toString())
        }

        val entries = timber.entries
        assertNotEquals(times, entries.size)
    }

    companion object {
        private const val TAG = "TimberTest"
    }

}
